packages/evm/test/validators/SignerValidator.t.sol

```diff
@@ -204,6 +207,54 @@ contract SignerValidatorTest is Test {
         validator.validate(boostId, incentiveId, claimant, claimData);
     }
 
+    function testValidate_FuzzValidInputs(
+        uint256 boostId,
+        uint256 incentiveId,
+        uint256 pk,
+        uint8 incentiveQuantity,
+        address claimant,
+        bytes memory incentiveData
+    ) public {
+        incentiveQuantity = uint8(bound(incentiveQuantity, 1, 7));
+        incentiveId = bound(incentiveId, 0, incentiveQuantity - 1);
+        pk = bound(pk, 1, _MAX_SECP256K_CURVE_VALUE);
+
+        bool[] memory signers_authorized = new bool[](1);
+        signers_authorized[0] = true;
+        address[] memory signers = new address[](1);
+        signers[0] = vm.addr(pk);
+
+        validator.setAuthorized(signers, signers_authorized);
+
+        bytes32 msgHash = validator.hashSignerData(boostId, incentiveQuantity, claimant, incentiveData);
+        bytes memory signature = _signHash(msgHash, pk);
+
+        ASignerValidator.SignerValidatorInputParams memory validatorData =
+            ASignerValidator.SignerValidatorInputParams(signers[0], signature, incentiveQuantity);
+        bytes memory claimData = abi.encode(IBoostClaim.BoostClaimData(abi.encode(validatorData), incentiveData));
+
+        validator.validate(boostId, incentiveId, claimant, claimData);
+    }
+
+    function testValidate_FuzzMaliciousSignature(
```

Comment:
We probably want to add another fuzz test for malicious signature generation vs signer generation and rename this but this one looks good.